pkg/util/validation/limits.go

```diff
@@ -88,6 +96,13 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&l.MaxGlobalSeriesPerMetric, "ingester.max-global-series-per-metric", 0, "The maximum number of active series per metric name, across the cluster. 0 to disable.")
 	f.IntVar(&l.MinChunkLength, "ingester.min-chunk-length", 0, "Minimum number of samples in an idle chunk to flush it to the store. Use with care, if chunks are less than this size they will be discarded.")
 
+	f.IntVar(&l.MaxMetricsPerQuery, "ingester.max-metrics-per-query", 5000, "The maximum number of metrics with metadata that a query can return.")
+	f.IntVar(&l.MaxMetadataPerQuery, "ingester.max-metadata-per-query", 10, "The maximum number of metadata per metric that a query can return.")
+	f.IntVar(&l.MaxLocalMetadataPerUser, "ingester.max-metadata-per-user", 10000, "The maximum number of active metrics with metadata per user, per ingester. 0 to disable.")
```

Comment:
At an average of ~55 bytes per metadata (a combination of HELP, TYPE, UNIT, METRIC NAME) we're talking of a maximum of about 55 x 10,000 x 50 which is ~27mb per tenant (plus the replication factor). 

String interning (which I have yet to do) seems like a great optimisation for this work because metadata will have a high overlap across tenants. Furthermore, metadata in typical scenarios should not go wide (what we call per metric). I have set per metric to 50 but in almost every case it should be 1 - so maybe I should lower than to 10? It would put us to around ~5mb per tenant (plus the replication factor)

Reply:
Querying values are much lower to protect queries but I think I will tweak them a bit more.